### ISSUE TITLE
Add information on purchase order modifications

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#5721]
We need a report that shows modifications to the purchase order.  This would show:

1. who modified the purchase order
2. when they modified the purchase order
3. what changed

I'm picturing something like the `inventory_log` that contains the details of any modifications to the purchase order.

**Comment by jmcameron** [comment]
Would this include details of deleted POs?

**Comment by jniles** [comment]
I don't think so.  Because purchase orders are not able to be removed after being confirmed, and they have no financial impact on the institution, it doesn't seem particularly useful to track that information.

**Comment by jniles** [comment]
The more I've thought about this, the more I think that we need to implement purchase quotes (https://github.com/IMA-WorldHealth/bhima/issues/177), instead of editing purchase orders.

The rational is the a purchase order should really be more of a contract, while the quote is something you can add/remove items and have price fluctuations.  Instead of having a "confirmed" purchase order, we ought to have a "quote" that anyone can modify, and convert it into a purchase order once all the modifications are made.
